spec/tools/fix_auth/auth_config_model_spec.rb

```diff
@@ -20,6 +20,37 @@
     ManageIQ::Password.clear_keys
   end
 
+  context "#recrypt" do
+    subject { FixAuth::FixMiqRequest }
+    let(:request) do
+      subject.create(
+        :type    => 'MiqProvisionRequest',
+        :options => YAML.dump(
+          :sysprep_password        => enc_old,
+          :sysprep_domain_password => enc_new
+        )
+      )
+    end
+
+    context "with the rare case where recryption succeeds but returns garbage" do
+      # NOTE: This legacy key only returns garbage specifically with the
+      #   built-in v2_key.dev and the plaintext string "password", which is
+      #   why it is redeclared here.  If this password is changed, a new
+      #   colliding legacy key will need to be found.
```

Comment:
Wow, that's crazy.

Reply:
To clarify a little, there are probably a huge number of colliding keys, and I just chose one that I knew would fail, but the key chosen is dependent on the other key _and_ the plaintext itself, which makes sense: Key A encrypts Text B into a specific binary pattern that happens to be decodable by Key C.